common/metrics/src/metric_server.rs

```diff
@@ -42,25 +63,64 @@ fn serve_metrics(req: Request<Body>) -> impl Future<Item = Response<Body>, Error
     future::ok(resp)
 }
 
-pub fn start_server<T: ToSocketAddrs>(to_addr: T) {
-    let addr: SocketAddr = to_addr
+fn serve_public_metrics(
+    req: Request<Body>,
+) -> impl Future<Item = Response<Body>, Error = hyper::Error> {
+    let mut resp = Response::new(Body::empty());
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, "/public_metrics") => {
+            //Create a different path to expose public metrics
+            let encoder = TextEncoder::new();
+            let buffer = encode_public_metrics(encoder);
+            *resp.body_mut() = Body::from(buffer);
+        }
+        _ => {
+            *resp.status_mut() = StatusCode::NOT_FOUND;
+        }
+    };
+
+    future::ok(resp)
+}
+
+pub fn start_server(host: String, port: u16, public: bool) {
```

Comment:
I'd suggest abstracting the public/private distinction here, and just pass in a list of whitelisted metrics instead. If the whitelist is not specified then no filtering is done (i.e. private) otherwise metrics are filtered (i.e. public).

Reply:
It's a bit too complicated to pass over the whitelist from here, so I kept the true / false...